test/ber_types/RealTest.cpp

```diff
@@ -0,0 +1,105 @@
+#include "fast_ber/ber_types/Real.hpp"
+
+#include "absl/types/optional.h"
+
+#include <catch2/catch.hpp>
+#include <limits>
+
+TEST_CASE("Real: Construction from double")
+{
+    std::initializer_list<double> test_vals{0.0,
+                                            -0.0,
+                                            std::numeric_limits<double>::infinity(),
+                                            -std::numeric_limits<double>::infinity(),
+                                            -9999999999.99999,
+                                            -27837.48,
+                                            -256.0,
+                                            -255.0,
+                                            -254.0,
+                                            -50.0,
+                                            -20.434,
+                                            -1.0,
+                                            0,
+                                            1.0,
+                                            255.022,
+                                            256.2343,
+                                            20.888,
+                                            50.00,
+                                            99999.9999,
+                                            27836.1232312,
+                                            27837.43645454,
+                                            27838.34324,
+                                            37699.34234,
+                                            999999999.9999};
+
+    for (double val : test_vals)
+    {
+        fast_ber::Real<>                                                   real1(val);
+        fast_ber::Real<fast_ber::Id<fast_ber::Class::context_specific, 2>> real2(real1);
+        fast_ber::Real<fast_ber::Id<fast_ber::Class::context_specific, 4>> real3;
+        absl::optional<fast_ber::Real<>>                                   real4;
+        absl::optional<fast_ber::Real<>>                                   real5(real1);
+
+        real3 = real2;
+        real4 = real3;
+
+        REQUIRE(real1 == val);
+        REQUIRE(real2 == val);
+        REQUIRE(real3 == val);
+        REQUIRE(real4 == val);
+        REQUIRE(real4 == val);
+    }
+}
+
+struct TestData
+{
+    std::vector<uint8_t> data;
+    double               value;
+};
+
+TEST_CASE("Real: Decoding")
+{
+    // todo:
+    // std::numeric_limits<double>::infinity()
+    // -std::numeric_limits<double>::infinity()
```

Comment:
Let's add these test cases in, and a few more from ASCII decimal, then we will be good to merge this PR

Reply:
Added todo cases and ISO6093 NR{1,2,3} cases